### PR TITLE
Added Fenwick Trees

### LINKED ISSUE #24: Add Fenwick Tree
#### Description of the problem

Fenwick trees are quite popular data structure and should be added to the package.

#### References/Other comments
https://en.wikipedia.org/wiki/Fenwick_tree

### PR DESCRIPTION
**Pull request by aravind-karthikeyan**
Added Fenwick Trees
Closes #24

### CODE REVIEW

File: pydatastructs/trees/fenwick_tree.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+__all__ = ['FenwickTree']
+class FenwickTree():
+    def __init__(self, array):
```
Comment (czgdp1807) [inline pydatastructs/trees/fenwick_tree.py]: We use `__new__` to create and initialise objects. See other modules and files for examples.

File: pydatastructs/trees/fenwick_tree.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        self.size = len(array) + 1
+        self.tree = [0] * (self.size + 1)
+        self.array = array
+        self.flag = [0] * (self.size-1)
+        for index in range(self.size-1):
+            self.update(index, array[index])
+    def update(self, index, value):
```
Comment (czgdp1807) [inline pydatastructs/trees/fenwick_tree.py]: An empty line should be added between two functions/methods.

File: pydatastructs/trees/tests/test_fenwick_tree.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from pydatastructs.trees import FenwickTree
+from pydatastructs.utils.raises_util import raises
+from copy import deepcopy
+def test_FenwickTree():
```
Comment (czgdp1807) [inline pydatastructs/trees/tests/test_fenwick_tree.py]: Add empty line.

**Comment by czgdp1807** [comment]
Please add doc strings to classes and methods. See other modules for examples.

**Comment by czgdp1807** [comment]
Are you participating through GSSoC, 2020?

**Comment by aravind-karthikeyan** [comment]
Made changes as you instructed

**Comment by aravind-karthikeyan** [comment]
> Are you participating through GSSoC, 2020?

No

**Comment by czgdp1807** [comment]
Thanks. Some minor changes are required. Please enable `Allow edit from maintainers` in right hand side of the page. Will push some commits to finalise.

**Comment by aravind-karthikeyan** [comment]
> Thanks. Some minor changes are required. Please enable `Allow edit from maintainers` in right hand side of the page. Will push some commits to finalise.

Allowed edit from maintainers

**Comment by aravind-karthikeyan** [comment]
Is there anything else need to be done?

**Comment by czgdp1807** [comment]
No. I am a bit busy to make some final changes to your PR. I will push the changes whenever I will find some time.

**Pull request merged**